src/tools/lib/python/particleCalorimeter.py

```diff
@@ -0,0 +1,245 @@
+        self.pos_yaw = pos_yaw
+        self.pos_pitch = pos_pitch
+
+        self.attribute_name = "calorimeter"
+
+        self.energy_range = np.linspace(self.min_energy,
+                                        self.max_energy,
+                                        self.num_bins_energy)
+
+        self.calo_data = np.zeros([self.num_bins_energy,
+                                   self.num_bins_pitch,
+                                   self.num_bins_yaw], dtype=np.float64)
+        self.hist_data = np.zeros([self.num_bins_energy, ], dtype=np.float64)
+
+        self.sim_unit = None
+
+    def load_step(self,
+                  step=5000):
+        """
+        Load an HDF5 File of a certain time step
+
+        Parameters
+        ----------
+        time : unsigned integer
+            PIConGPU output time step
+        """
+        # convert the time step into a string
+        self.stepStr = str(int(step))
+
+        self.calo_file = self.sim + "/simOutput/" \
+            + self.species_name + "_" \
+            + self.attribute_name + "/" \
+            + self.species_name + "_" \
+            + self.attribute_name + "_" \
+            + self.stepStr + "_0_0_0.h5"
+
+        if not os.path.isfile(self.calo_file):
+            print(
+                "Error: The output HDF5 file " +
+                self.calo_file +
+                " does not exist."
+            )
+            return
+
+        f = h5py.File(self.calo_file, 'r')
+
+        self.step = step
+        # calorimeter data
+        self.calo_data = np.array(f['/data/' +
+                                    self.stepStr + "/" +
+                                    self.attribute_name])
+        # simulation unit
+        self.sim_unit = f['/data/' +
+                          self.stepStr + "/" +
+                          self.attribute_name].attrs['unitSI']
+
+        # cleanup
+        f.close()
+
+    def energy_histogram(self):
+        """
+        Creates an energy histogram that can be compared to the output of the
+        `energyHistogram` plugin output. The data is therefore summed
+        over the yaw and pitch angles.
+        """
+        self.hist_data = np.sum(self.calo_data, axis=(1, 2))
+
+    def plot_histogram(self,
+                       with_outliers=False,
+                       norm=False):
+        """
+        Parameters
+        ----------
+        with_outliers : boolean
+            plots also underflow and overflow bins if ``True``
+        norm : boolean
+            normalizes the histogram to the total number of counts
+            in all bins if ``True``
+        """
+        # unit conversion: Joules to eV
+        J2EV = 1.602e-19
+        # unit conversion: simUnit to EV
+        simUnitEV = self.sim_unit / J2EV
+
+        figHist = plt.figure(0)
+        axHist = figHist.add_subplot(111)
+
+        # absolute histogram data [photon count]
+        abs_hist_data = self.hist_data * \
+            simUnitEV / 1e3 / self.energy_range
```

Comment:
yes it is, since the PIConGPU input parameters for this plugin are in units of keV